### ISSUE TITLE
gnome-shell 4.0: Error: No property border_width on GtkFrame

### ISSUE THREAD

**Issue by motoshkov** [paradoxxxzero/gnome-shell-system-monitor-applet#659]
On Fedora-34:
version:
gnome-shell-extension-system-monitor-applet-38-10.20200503git7f8f0a7.fc34.noarch

Error: No property border_width on GtkFrame

Stack trace:
  _init/Gtk.Widget.prototype._init@resource:///org/gnome/gjs/modules/core/overrides/Gtk.js:45:40
  SystemMonitor@/usr/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/prefs.js:141:22
  SystemMonitor_App/<@/usr/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/prefs.js:295:38
  SystemMonitor_App@/usr/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/prefs.js:294:23
  buildPrefsWidget@/usr/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/prefs.js:367:18
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:209:40
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17

**Comment by marxin** [comment]
I see the same, I would rename the issue to "Support GNOME 40" instead.

**Comment by wuqso** [comment]
I'm also waiting for the update for the gnome shell 40.

**Comment by sallyahaj** [comment]
Same issue with the extension on Fedora 34 beta, on Gnome 40

**Comment by ojmarcelino** [comment]
Exactly same issue on Arch with Gnome 40

**Comment by chrisspen** [comment]
Has anyone tried testing the changes in PR #678?

**Comment by wuqso** [comment]
go to ~/.local/share/gnome-shell/extensions into your extension folder, edit the "metadata.json" and add an "40.0" entry into the "shell-version" array.
After a relogin or restart you have your extensions back.

**Comment by ojmarcelino** [comment]
actually, I've disable the check by overriding it with:
$ gsettings set org.gnome.shell disable-extension-version-validation "true"

**Comment by sallyahaj** [comment]
> actually, I've disable the check by overriding it with:
> $ gsettings set org.gnome.shell disable-extension-version-validation "true"

Does it working as expected?

**Comment by ojmarcelino** [comment]
> Does it working as expected?

Pretty much, yes, it works as a "nuclear" option, since it don't check any of user's extensions list, but, in my case, happened a few "glitches" in one or two features from other extensions, which varies the "symptoms". Anyway, mostly of them are basically api and calls from gtk3 to gtk4.

**Comment by p91paul** [comment]
> Has anyone tried testing the changes in PR #678?

I did and reported an issue on the PR comment section. For most users should work fine though, it's just a bit annoying for smaller screen (the preferences window gets way larger).

**Comment by ojmarcelino** [comment]
> > Has anyone tried testing the changes in PR #678?
> 
> I did and reported an issue on the PR comment section. For most users should work fine though, it's just a bit annoying for smaller screen (the preferences window gets way larger).

But that could be postponed to some further dev, or even if someone want to redesign the extension. IMHO the smaller screen crowd (as you said these days, if any), these users should rely on other ways to configure their screens, like rescale, font size, nvidia viewport, anyway, not an issue (again, imho, sorry if I'm bothered some UX people), to me it sounds more like a new version ( to don't say something like.... have you ever considered using top?)

**TL;DR**: working flawlessly right now in Gnome 40.1 over Arch Linux here.

**Comment by aluedtke7** [comment]
> **TL;DR**: working flawlessly right now in Gnome 40.1 over Arch Linux here.

I'm using Manjaro and having still problems with this extension to get the preferences working.

I'm using the git repo (branch master, HEAD). When I execute a "make install" I can see that the extension was copied to .local/share/gnome-shell/. I then logout/login and the extension change is detected and I have to logout/login again to apply the changes. But the preferences can't be open. Same error as before.

Have you done something else to get it working?

**Comment by aluedtke7** [comment]
I could fix it myself with the following workflow:

```
make install
make zip-file VERSION=40
gnome-extensions install ~/path_to_the_git_repo/_build/system-monitor@paradoxxx.zero.gmail.com_v40.zip
```

Now the preferences are working.

**Comment by ojmarcelino** [comment]
> Have you done something else to get it working?

So sorry, but don't know. Despite, afaik, Manjaro is derived from Arch, it ain't the same thing, imho, got a lot of weird stuff on Manjaro which works just fine on Arch. I've just followed the wikis and this issues instructions. That's all folks...

**Comment by MahbodHastam** [comment]
> actually, I've disable the check by overriding it with:
> $ gsettings set org.gnome.shell disable-extension-version-validation "true"

@ojmarcelino 
It doesn't work for me.
I have the same issue with some of the extensions.

**Comment by ojmarcelino** [comment]
> I have the same issue with some of the extensions.

one may use gsettings or dconf, maybe the easy way is using dconf-editor (pacman -S dconf-editor), then run it and go to org/gnome/shell and use the slider to switch enable/disable, then press the F2 and "r", without quotes, to reload the gnome, or go to a terminal and run the systemctl restart gdm, reboot

hope that helps
